silverback/main.py

```diff
@@ -593,22 +597,20 @@ def on_(
 
             return self.broker_task_decorator(TaskType.NEW_BLOCK, container=container)
 
-        elif isinstance(container, ContractEvent) and isinstance(
-            container.contract, ContractInstance
-        ):
-            key = container.contract.address
-
-            if new_block_timeout is not None:
-                if key in self.poll_settings:
-                    self.poll_settings[key]["new_block_timeout"] = new_block_timeout
-                else:
-                    self.poll_settings[key] = {"new_block_timeout": new_block_timeout}
-
-            if start_block is not None:
-                if key in self.poll_settings:
-                    self.poll_settings[key]["start_block"] = start_block
-                else:
-                    self.poll_settings[key] = {"start_block": start_block}
+        elif isinstance(container, ContractEvent):
+            if isinstance(container.contract, ContractInstance):
+                key = container.contract.address
+                if new_block_timeout is not None:
+                    if key in self.poll_settings:
+                        self.poll_settings[key]["new_block_timeout"] = new_block_timeout
+                    else:
+                        self.poll_settings[key] = {"new_block_timeout": new_block_timeout}
+
+                if start_block is not None:
+                    if key in self.poll_settings:
```

Comment:
style: Several spots can be cleaned up using `.setdefault()`, e.g.:

```
self.poll_settings.setdefault(key, {})
self.poll_settings[key]["start_block"] = start_block
```

Reply:
tbh I've been trying to migrate this code into Ape, under `ProviderAPI.iter_logs`/`ProviderAPI.iter_blocks` methods that can go figure out how to configure using websocket or http polling based on how it is configured to operate (if it has websockets/IPC or not)